### PR TITLE
Feature/dev 4 Header module

### PR DESCRIPTION
**Pull request by savelievamareena**

### CODE REVIEW

File: src/modules/header/components/navigation/navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            styles={{
+                header: { padding: "10px", backgroundColor: "#2E2E2E" },
+                body: { padding: 0 },
+            }}
+        >
+            <div className={styles.nav_header}>
+                <CloseOutlined onClick={onClose} className={styles.close_icon} />
+            </div>
+            <div className={styles.drawerContent}>
+                <div className={classNames(styles.menu_item, styles.border_bottom, styles.active)}>
```
Comment (AAKlimkov) [inline src/modules/header/components/navigation/navigation.tsx]: i think bad idea to use div for links

File: src/modules/header/components/profile-picture/profile-picture.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    useEffect(() => {
+        // get and setAvatarLink avatar link for the current user
+        setAvatarLink("");
+    });
+
+    return avatarLink !== "" ? (
+        <Image width={40} src={avatarLink} />
+    ) : (
+        <div className={styles.profile_picture_filler}></div>
```
Comment (AAKlimkov) [inline src/modules/header/components/profile-picture/profile-picture.tsx]: mb we can use same tags?

File: src/modules/header/dropdown-languages/dropdown-languages.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    const [isLanguagesOpen, setLanguagesOpen] = useState(false);
+    const [language, setLanguage] = useState<LanguageType>("EN");
+
+    const languages: MenuProps["items"] = [
+        {
+            label: <span style={{ padding: "12px", fontSize: "18px" }}>English</span>,
+            key: "EN",
+        },
+        {
+            label: <span style={{ padding: "12px", fontSize: "18px" }}>Deutsch</span>,
```
Comment (AAKlimkov) [inline src/modules/header/dropdown-languages/dropdown-languages.tsx]: as i know we are not going translate to german

File: package.json [hunk 4]
```diff
@@ -13,9 +13,11 @@
         "prepare": "husky install"
     },
     "dependencies": {
-        "@hookform/resolvers": "^3.4.2",
```
Comment (shev7) [inline package.json]: why was this removed?

File: src/modules/header/components/navigation/navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                </div>
+                <div className={styles.menu_item}>
+                    <ContactsOutlined style={{ fontSize: "20px" }} /> Positions
+                </div>
+                <div className={styles.menu_item}>
+                    <RiseOutlined style={{ fontSize: "20px" }} /> Skills
+                </div>
+                <div className={styles.menu_item}>
+                    <GlobalOutlined style={{ fontSize: "20px" }} /> Languages
+                </div>
```
Comment (shev7) [inline src/modules/header/components/navigation/navigation.tsx]: Icons should be wrapped into links, or if it is a button then into button tag

File: src/modules/header/dropdown-languages/dropdown-languages.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import { Dropdown, DropdownProps, MenuProps } from "antd";
+import { CaretDownOutlined, CaretUpOutlined } from "@ant-design/icons";
+import classNames from "classnames";
+import styles from "../header.module.css";
+
+type LanguageType = "EN" | "DE" | "RU";
```
Comment (shev7) [inline src/modules/header/dropdown-languages/dropdown-languages.tsx]: use enum values

File: src/modules/header/dropdown-languages/dropdown-languages.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                {isLanguagesOpen ? (
+                    <CaretUpOutlined
+                        className={classNames(styles.header_icons, styles.white, styles.small)}
+                    />
+                ) : (
+                    <CaretDownOutlined
+                        className={classNames(styles.header_icons, styles.white, styles.small)}
+                    />
+                )}
+            </div>
```
Comment (shev7) [inline src/modules/header/dropdown-languages/dropdown-languages.tsx]: maybe ant has some component for this, instead of just div

and also aboid using inline styles

also ```classNames``` can be named as ```cn```, it will be shorter

File: src/modules/header/dropdown-languages/dropdown-languages.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import { Dropdown, DropdownProps, MenuProps } from "antd";
+import { CaretDownOutlined, CaretUpOutlined } from "@ant-design/icons";
+import classNames from "classnames";
+import styles from "../header.module.css";
+
+type LanguageType = "EN" | "DE" | "RU";
+
+const DropdownLanguages = () => {
```
Comment (shev7) [inline src/modules/header/dropdown-languages/dropdown-languages.tsx]: Strange name
It will be better to name it as ```LanguagesDropdown``` or ```LanguagesSelect```, something like that

File: src/modules/header/dropdown-languages/dropdown-languages.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        setLanguage(key as LanguageType);
+        setLanguagesOpen(false);
+    };
+
+    const langMenuProps: MenuProps = {
+        items: languages,
+        selectable: true,
+        defaultSelectedKeys: [language],
+        onClick: handleSelectLanguage,
+    };
```
Comment (shev7) [inline src/modules/header/dropdown-languages/dropdown-languages.tsx]: put it to the jsx, no need to create separate object

File: src/modules/header/dropdown-user/dropdown-user.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import { ProfilePicture } from "../components/profile-picture";
+import { Dropdown, MenuProps } from "antd";
+import { LogoutOutlined, SettingOutlined, UserOutlined } from "@ant-design/icons";
+
+const DropdownUser = () => {
```
Comment (shev7) [inline src/modules/header/dropdown-user/dropdown-user.tsx]: also starange name, consider another one

File: src/modules/header/dropdown-user/dropdown-user.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+                </span>
+            ),
+            key: "3",
+        },
+    ];
+
+    const profileMenuProps: MenuProps = {
+        items: profileMenu,
+        selectable: false,
+    };
```
Comment (shev7) [inline src/modules/header/dropdown-user/dropdown-user.tsx]: the same

File: src/modules/header/header.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import classNames from "classnames";
+import { Flex, Layout } from "antd";
+import { Navigation } from "@/modules/header/components/navigation";
+import { GlobalOutlined, MenuOutlined } from "@ant-design/icons";
+import { DropdownLanguages } from "./dropdown-languages";
+import { DropdownUser } from "./dropdown-user";
+import styles from "./header.module.css";
+
+const Header = () => {
+    const { Header: AntdHeader } = Layout;
```
Comment (shev7) [inline src/modules/header/header.tsx]: use it as ```<Layout.Header/>``` in the jsx

File: src/modules/header/header.tsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    const [isDrawerOpen, setDrawerOpen] = useState(false);
+
+    return (
+        <>
```
Comment (shev7) [inline src/modules/header/header.tsx]: you don't need react fragment here

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/modules/header/components/navigation/navigation.tsx (hunk 1 above)
Comment (savelievamareena) [reply src/modules/header/components/navigation/navigation.tsx]: I haven't yet had the access to router at that time, so Links without routes might give errors, routing in navigation menu will be implemented in DEV-12/connect-sidebar-navigation-with-the-router issue

File: src/modules/header/dropdown-languages/dropdown-languages.tsx (hunk 3 above)
Comment (savelievamareena) [reply src/modules/header/dropdown-languages/dropdown-languages.tsx]: ok

File: src/modules/header/components/navigation/navigation.tsx (hunk 1 above)
Comment (shev7) [reply src/modules/header/components/navigation/navigation.tsx]: not a problem, issue will be fixed later
for now in can rerdirect to current page

File: src/modules/header/components/navigation/navigation.tsx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    FolderOutlined,
+    GlobalOutlined,
+    HomeFilled,
+    RiseOutlined,
+    UserOutlined,
+} from "@ant-design/icons";
+import styles from "./navigation.module.css";
+import { Link } from "react-router-dom";
+
+interface NavigationTypes {
```
Comment (shev7) [inline src/modules/header/components/navigation/navigation.tsx]: NavigationProps

File: src/modules/header/languages-select/languages-select.tsx [hunk 15]
```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { Dropdown, DropdownProps, Flex, MenuProps } from "antd";
+import { CaretDownOutlined, CaretUpOutlined } from "@ant-design/icons";
+import classNames from "classnames/bind";
+import styles from "../header.module.css";
+
+export enum LanguageType {
+    EN = "EN",
+    RU = "RU",
+}
```
Comment (shev7) [inline src/modules/header/languages-select/languages-select.tsx]: this enum already exists, was created by @AAKlimkov

**Review by shev7 — APPROVED** [review]

**Review by RC-3222 — APPROVED** [review]

**Review by AAKlimkov — APPROVED** [review]
